tests/integration/host/test_remote.py

```
@@ -0,0 +1,119 @@
+# ruff: noqa: SLF001
+
+import os
+import re
+
+import pytest
+from python_on_whales import DockerException
+
+pytestmark = pytest.mark.skipif(
+    not os.environ.get("CI"),
+    reason="Only run integration tests in CI, unless forced with 'CI' env var",
+)
+
+
+def test_remote(slurm_host, mocker, compose_containers):
+    host = slurm_host("johndoe")
+
+    connection_spy = mocker.spy(host.connection, "run")
+
+    stdout, stderr, return_code = host.execute(["test", "-e", "/home/johndoe"])
+    connection_spy.assert_called_once()
+    args, kwargs = connection_spy.call_args
+    # Here we remove multiple spaces as
+    assert re.sub(r"\s+", " ", args[0]).strip() == "bash -l -c 'test -e /home/johndoe'"
+    assert stdout == ""
+    assert stderr == ""
+    assert return_code == 0
+
+    docker_client = compose_containers
+    # Make sure the directory does not exist yet
+    with pytest.raises(DockerException):
+        docker_client.execute(
+            "qtoolkit_slurm", ["test", "-e", "/home/johndoe/remote_tests"]
+        )
+
+    assert host.mkdir("/home/johndoe/remote_tests") is True
+    # Now we check that the directory was created
+    output = docker_client.execute(
+        "qtoolkit_slurm", ["test", "-d", "/home/johndoe/remote_tests"], user="johndoe"
+    )
+    assert output == ""
+    assert host.mkdir("/home/johndoe/remote_tests", exist_ok=False) is False
```

This is not about the tests, but looking at this test makes me wonder if this makes sense at all in the Host object. In the normal python code (e.g. with Path.mkdir) you are able to tell apart a failure in creating the folder from one where the folder already exists. But here we are left with a `False` output and you cannot know if the folder actually exists or not. I wonder if we should raise an exception instead? In jobflow-remote it is the same and I think that even there this is somewhat impractical: if the runner fails to create the folder you just get an error message stating that the folder is not created, but there is no way of knowing why (connection issue? file system permission?) because you just get a boolean in output. I suppose I(we?) did this not to have to deal with all the potential different errors of different kinds of host, but now I am questioning if we should raise exceptions instead. 

In any case not in this PR.

---

Sure, we can think of that in another PR.